Refs #373 -- Added Model._is_pk_set(). (#18450)
django/django: docs/ref/models/instances.txt

    since they are yet to be saved. Instances fetched from a ``QuerySet``
    will have ``adding=False`` and ``db`` set to the alias of the associated
    database.

``_is_pk_set()``
----------------

.. method:: Model._is_pk_set()

> The ``_is_pk_set()`` method decides whether the ``pk`` property has been _set_.

Reviewer: ```suggestion
The ``_is_pk_set()`` method returns whether the ``pk`` property is defined or not.
```

> The ``pk`` property is considered set if it's not ``None``.

Reviewer: Do we need to clarify this? (honest question)
In a way, we are adding this helper to abstract the users from knowing/understanding when a pk is set, so maybe we should remove it (perhaps this is a case where "less is more").
What do you think?
